feat(de55): add error modes; validate DE55 inputs

dertin opened the pull request:
- Validate incomplete/truncated DE55 data in `_icc_to_dict`.
- Introduce `field_error_handling` with `ERROR` (default) and `WARN`.
- Add comprehensive unit tests for invalid inputs and correct tag extraction.

Example:
```python
from cardutil.config import config

custom_config = config["bit_config"].copy()
custom_config["55"]["field_error_handling"] = "WARN"
...
reader = IpmReader(f, blocked=True, iso_config=custom_config)
```

dertin commented:
Hi @adelosa, when reading an IPM file with IpmReader(f, blocked=True, encoding='cp500'), we sometimes get invalid DE55 data due to an issue from an acquirer. I added a setting to choose parser behavior: either fail with a parsing error or ignore the invalid DE55 and log a warning. I had to do this as a workaround to fix a production issue. 
What do you think about this approach?

EmilianoQS commented:
Hi @adelosa, thank you for your excellent work! We've found your library to be very useful.
We're using your library in a production environment, and as @dertin mentioned above, we're encountering issues with some acquirers.
This fix implements a "soft parse" (does not generate a file-level exception) for DE55 in the IPM files. We consider this fix suitable for production, as DE55 is already validated during transaction authorization, and a failed validation in the POST process has no impact, since the transaction is already confirmed (present in the IPM file).

Thanks!

adelosa commented:
Thanks so much for your pull request. This is the first one I have ever received for this project so worthy of celebration.

The pull request looks really good.

I am wondering if introducing a new field parameter that could be applied to all fields is the right approach. 

I would just add details on how the DE55 field processor handles errors to field_processor_config rather than introducing a new field config. Thats what this field is for.

So that would be that in the config for DE55, the config could look something like:
```
"55": {
            "field_name": "ICC system related data",
            "field_type": "LLLVAR",
            "field_length": 255,
            "field_processor": "ICC",
            "field_processor_config": "on_error=WARN"
        },
```
or something similar.. Open to ideas on how field_processor_config should be structured.

If we were introducing new error handling across all fields, then your approach would definitely be something I would be open to.

I would also be happy if WARN was the default for the package.

Hoping you can update the PR to accommodate this request. If you have any questions, let me know.

dertin commented:
Hi @adelosa 
I’ve pushed a new commit addressing your review. DE55 error handling is now configured via `field_processor_config` (k=v;...), supporting `on_error=ERROR|WARN`, with `on_error=WARN` as the default. Docs and tests updated. When you have a moment, please re-review; feel free to edit this branch directly before merge.

adelosa commented:
Thanks Guillermo for the updates.. Will release a new version with this change included.
